### PR TITLE
Upgrade to rustls v0.22

### PR DESCRIPTION
**Pull request by s-arash**
I also deleted some tests that were not relevant to us.

### CODE REVIEW

File: rustls-mbedcrypto-provider/src/tls12.rs [hunk 1]
```diff
@@ -271,6 +281,11 @@ impl MessageEncrypter for GcmMessageEncrypter {
 
         Ok(OpaqueMessage::new(msg.typ, msg.version, payload))
     }
+
+    fn encrypted_payload_len(&self, payload_len: usize) -> usize {
+        const TAG_LEN: usize = 16;
+        payload_len + GCM_EXPLICIT_NONCE_LEN + TAG_LEN
```
Comment (Taowyoo) [inline rustls-mbedcrypto-provider/src/tls12.rs:287]:
```suggestion
        payload_len + GCM_EXPLICIT_NONCE_LEN + aead::TAG_LEN
```

**Review by Taowyoo — CHANGES_REQUESTED** [review]
Just a small change needed. Also it seems there is some syntax issue complained by clippy.

File: rustls-mbedpki-provider/src/tests_common.rs [hunk 2]
```diff
@@ -12,24 +12,22 @@ use core::{
...
 
 /// Get a private key from the contents of a pem file
 pub(crate) fn get_key(bytes: &[u8]) -> PrivateKeyDer {
     let value = rustls_pemfile::pkcs8_private_keys(&mut io::BufReader::new(bytes))
-        .unwrap()
-        .into_iter()
         .next()
+        .unwrap()
         .unwrap();
     PrivateKeyDer::from(PrivatePkcs8KeyDer::from(value))
```
Comment (Taowyoo) [inline rustls-mbedpki-provider/src/tests_common.rs:32]:
```suggestion
     PrivateKeyDer::from(value)
```

File: rustls-mbedpki-provider/src/tests_common.rs [hunk 3]
```diff
@@ -12,24 +12,22 @@ use core::{
     ops::{Deref, DerefMut},
 };
 
-use pki_types::{CertificateDer, PrivateKeyDer, PrivatePkcs8KeyDer, UnixTime};
+use pki_types::{CertificateDer, PrivateKeyDer, PrivatePkcs8KeyDer, ServerName, UnixTime};
```
Comment (Taowyoo) [inline rustls-mbedpki-provider/src/tests_common.rs:15]:
```suggestion
use pki_types::{CertificateDer, PrivateKeyDer, ServerName, UnixTime};
```
Remove unused type, see  comment below

File: rustls-mbedpki-provider/src/tests_common.rs (hunk 2 above)
Comment (s-arash) [reply rustls-mbedpki-provider/src/tests_common.rs:32]: Good catch. Thanks.

**Pull request merged**